src/test/groovy/org/prebid/server/functional/tests/privacy/GppAuctionSpec.groovy

```diff
@@ -0,0 +1,180 @@
+package org.prebid.server.functional.tests.privacy
+
+import com.iab.gpp.encoder.section.TcfCaV1
+import org.prebid.server.functional.model.request.auction.BidRequest
+import org.prebid.server.functional.model.request.auction.User
+import org.prebid.server.functional.model.response.auction.ErrorType
+import org.prebid.server.functional.tests.BaseSpec
+import org.prebid.server.functional.util.PBSUtils
+import org.prebid.server.functional.util.privacy.CcpaConsent
+import org.prebid.server.functional.util.privacy.GppConsent
+
+import static org.prebid.server.functional.util.privacy.CcpaConsent.Signal.ENFORCED
+
+class GppAuctionSpec extends BaseSpec {
+
+    def "PBS should populate gdpr to 1 when regs.gdpr is not specified and gppSid contains 2"() {
+        given: "Default bid request with gppSid and without gdpr"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gdpr = null
+            regs.gppSid = [2]
+        }
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain regs.gdpr"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gdpr == 1
+    }
+
+    def "PBS should populate gdpr to 0 when regs.gdpr is not specified and gppSid not contains 2"() {
+        given: "Default bid request with gppSid and without gdpr"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gdpr = null
+            regs.gppSid = [PBSUtils.getRandomNumberWithExcept(2)]
+        }
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain regs.gdpr"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        assert bidderRequest.regs.gdpr == 0
+    }
+
+    def "PBS should emit warnings when regs.gpp_sid contains 2 and regs.gdpr isn't 1"() {
+        given: "Default bid request with gppSid and gdpr"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            regs.gdpr = 0
+            regs.gppSid = [2]
+        }
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bid response should contain warning"
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.code } == [999]
+        assert response.ext?.warnings[ErrorType.PREBID]?.collect { it.message } ==
+                ["GPP scope does not match TCF2 scope"]
+    }
+
+    def "PBS should emit warnings when regs.gpp_sid not contains 2 and regs.gdpr isn't 0"() {
```

Comment:
Same. What should happen to the `regs.gdpr` in this case?

Reply:
Same